lib/util/cache/repository/types.ts

```diff
@@ -1,4 +1,8 @@
 import type { PackageFile } from '../../../modules/manager/types';
+import type {
+  ApiPageCache,
+  GhRestPr,
+} from '../../../modules/platform/github/types';
```

Comment:
i know it's type only, but still circular references. 🤔 any better solution?

Reply:
BTW, how do you detect circular type import dependencies?

Reply:
i think we currently can't detect it, as they are gone after transpile 🤔

Reply:
It seems like solution will create more mess than the original problem.
We have extract all the parts that are used in repo cache as well as in a bunch of other places. In another words, along with `types.ts` files we also would create something like `other-types.ts` for each folder using repo cache.

Reply:
or only use some base type and cast in platform code?

Reply:
Don't we risk to get actual data structure mismatch that won't be handled by type system?

Reply:
One more alternative is to take advantage of structural type equivalence, but it means we have to duplicate all the types in repo cache 🤷‍♂️

Reply:
Let's discuss this problem in separate issue, maybe? I think it's not the problem of this particular PR.

Reply:
sure, but to resolve circular references i usually move platform types to types folders, so they can more easily referenced.

i don't think we get structural issues in cache, as only github platform needs to know the type details. so i suggest to have some `github: Record<string, unknown>` on cache object and the platform code will cast it to the known interface.

Reply:
Okay, just fixed it